src/Network.cpp

```diff
@@ -890,11 +892,15 @@ void compare_net_outputs(std::vector<float>& data,
     constexpr float relative_error = 5e-2f;
     for (auto idx = size_t{0}; idx < data.size(); ++idx) {
         auto err = relative_difference(data[idx], ref[idx]);
-        if (err > relative_error) {
-            printf("Error in OpenCL calculation: expected %f got %f (%lli"
+        if (display_only) {
+            myprintf("compare_net_outputs %s idx %d data %f ref %f err=%f\n",
+                info.c_str(), idx, data[idx], ref[idx], err);
+        } else if (err > relative_error) {
+            almost_equal = false;
+            myprintf("Error in OpenCL calculation: expected %f got %f (%lli"
```

Comment:
Shouldn't this be myprintf_so?

Reply:
I sent it to stderr to avoid messing up the UCI protocol. The idea is to let the program to continue because it hasn't hit the num_expansions < min_correct_expansions limit yet, but also log it so the users can debug what's going on.

Maybe I could send it to stdout with `info string <error message>`? I think that would follow UCI debug protocol. It might get a little verbose if we send a bunch of debug messages with `info string` in front of them instead of just using stderr. What do you think?